src/Components/GetWeatherData.js

```diff
@@ -0,0 +1,74 @@
+// ***********************
+// *** Modules & Hooks ***
+// ***********************
+import { useEffect, useState } from "react";
+import axios from "axios";
+import { citiesList } from "../modules/citiesList";
+
+// The function below (getCityID) takes in a city name as a string (i.e. "Halifax", "Toronto", etc) and returns the unique ID for that city that we need for the API call.
+const getCityID = (city) => {
+  const findCityID = citiesList.find(e => e.city === city);
+  return findCityID.id;
+}
+
+// API key and API Url for the axios call
+const apiKey = "oU28MIP77GV30LbR2diTL2ACEcTuAWoZ";
+const baseURL = "https://dataservice.accuweather.com/currentconditions/v1/";
+
+// API call to get weather data
+const GetWeatherData = (props) => {
+  const [weather, setWeather] = useState([]);
+  const [error, setError] = useState("");
+  const [loading, setLoading] = useState(true);
+
+// find ID of city for API search, add to searchURL
+  const ID = getCityID(props.cityName);
+  const searchURL = `${baseURL}${ID}`; 
+  
+// API axios call inside a useEffect hook, with 1 dependency. This will run anytime the city name changes (props.cityName).
+  useEffect(() => {
+    axios({
+      url: searchURL,
+      method: "GET",
+      params: {
+        apikey: apiKey,
+        language: "en-us",
+        details: false,
+      },
+    }).then(response => {
+      // store response data in variables for temperature, tempurature units (celsius or fahrenheit), and brief text description of the weather (i.e. "Light rain", "Mostly cloudy", etc).
+      const responseData = response.data;
+      const temp = responseData[0].Temperature.Metric.Value;
```

Comment:
are you going to use this variable a lot? you probably don't need to store it in a variable otherwise